usvm-ts/src/main/kotlin/org/usvm/TSMachine.kt

```diff
@@ -1,11 +1,126 @@
+import org.jacodb.panda.dynamic.ets.model.EtsFile
+import org.jacodb.panda.dynamic.ets.model.EtsMethod
+import org.usvm.ps.createPathSelector
+import org.usvm.state.TSMethodResult
 import org.usvm.state.TSState
+import org.usvm.statistics.*
+import org.usvm.statistics.collectors.AllStatesCollector
+import org.usvm.statistics.collectors.CoveredNewStatesCollector
+import org.usvm.statistics.collectors.TargetsReachedStatesCollector
+import org.usvm.statistics.distances.CfgStatistics
+import org.usvm.statistics.distances.CfgStatisticsImpl
+import org.usvm.statistics.distances.PlainCallGraphStatistics
+import org.usvm.stopstrategies.createStopStrategy
+import kotlin.time.Duration.Companion.seconds
 
 class TSMachine(
+    private val project: EtsFile,
     private val options: UMachineOptions
 ) : UMachine<TSState>() {
+
+    private val typeSystem = TSTypeSystem(typeOperationsTimeout = 1.seconds, project)
+    private val components = TSComponents(typeSystem, options)
+    private val ctx: TSContext = TSContext(components)
+    private val applicationGraph: TSApplicationGraph = TSApplicationGraph(project)
+    private val interpreter: TSInterpreter = TSInterpreter(ctx, applicationGraph)
+    private val cfgStatistics = CfgStatisticsImpl(applicationGraph)
+
+    fun analyze(
+        methods: List<EtsMethod>,
+        targets: List<TSTarget> = emptyList(),
+    ): List<TSState> {
+        val initialStates = mutableMapOf<EtsMethod, TSState>()
+        methods.forEach { initialStates[it] = interpreter.getInitialState(it, targets) }
+
+        val methodsToTrackCoverage =
+            when (options.coverageZone) {
+                CoverageZone.METHOD,
+                CoverageZone.TRANSITIVE,
+                -> methods.toSet()
+                // TODO: more adequate method filtering. !it.isConstructor is used to exclude default constructor which is often not covered
+                CoverageZone.CLASS -> methods.mapNotNull { method ->
+                    project.getMethodBySignature(method.signature)
+                }.toSet() + methods
+            }
+
+        val coverageStatistics: CoverageStatistics<EtsMethod, EtsStmt, TSState> = CoverageStatistics(
+            methodsToTrackCoverage,
+            applicationGraph
+        )
+
+        val callGraphStatistics: PlainCallGraphStatistics<EtsMethod> =
+            when (options.targetSearchDepth) {
+                0u -> PlainCallGraphStatistics()
+                else -> TODO("Unsupported yet")
+            }
+
+        val loopTracker = TSLoopTracker()
+        val timeStatistics = TimeStatistics<EtsMethod, TSState>()
+        val transparentCfgStatistics = transparentCfgStatistics()
+
+        val pathSelector = createPathSelector(
+            initialStates,
+            options,
+            applicationGraph,
+            timeStatistics,
+            { coverageStatistics },
+            { transparentCfgStatistics },
+            { callGraphStatistics },
+            { loopTracker }
```

Comment:
We are not sure about loop tracker, lets not include so complicated parts in the initial commits